test/DGmethods/compressible_Navier_Stokes/MWE_external_topography.jl

```diff
@@ -0,0 +1,267 @@
+                       W_g(t, x, y, z, Val(3)))
+    state.ρe = E_g(t, x, y, z, Val(3))
+end
+
+function mms3_source!(source::Vars, state::Vars, aux::Vars, t::Real)
+    x,y,z = aux.coord.x, aux.coord.y, aux.coord.z
+    source.ρ  = Sρ_g(t, x, y, z, Val(3))
+    source.ρu = SVector(SU_g(t, x, y, z, Val(3)),
+                        SV_g(t, x, y, z, Val(3)),
+                        SW_g(t, x, y, z, Val(3)))
+    source.ρe = SE_g(t, x, y, z, Val(3))
+end
+
+
+function run(mpicomm, ArrayType, dim, topl, warpfun, N, timeend, DFloat, dt)
+    
+    grid = DiscontinuousSpectralElementGrid(topl,
+                                            FloatType = DFloat,
+                                            DeviceArray = ArrayType,
+                                            polynomialorder = N,
+                                            meshwarp = warpfun,
+                                            )
+    
+    model = AtmosModel(ConstantViscosityWithDivergence(DFloat(μ_exact)),DryModel(),NoRadiation(),
+                       mms3_source!, InitStateBC(), mms3_init_state!)
+    
+    dg = DGModel(model,
+                 grid,
+                 Rusanov(),
+                 DefaultGradNumericalFlux())
+
+    param = init_ode_param(dg)
+
+    Q = init_ode_state(dg, param, DFloat(0))
+    
+    lsrk = LSRK54CarpenterKennedy(dg, Q; dt = dt, t0 = 0)
+    
+    mkpath("./vtk-mesh")
+    step = [0]
+    statenames = ("RHO", "U", "V", "W", "E")
+    cbvtk = GenericCallbacks.EveryXSimulationSteps(1) do (init=false)
+        outprefix = @sprintf("./vtk-mesh/stretched_grid")
+        
+        @debug "doing VTK output" outprefix
+        writevtk(outprefix, Q, dg, statenames)
+        step[1] += 1
+        nothing
+    end
+    solve!(Q, lsrk, param; timeend=timeend, callbacks=(cbvtk, ))
+    
+end
+
+using Test
+let
+    MPI.Initialized() || MPI.Init()
+    mpicomm = MPI.COMM_WORLD
+    ll = uppercase(get(ENV, "JULIA_LOG_LEVEL", "INFO"))
+    loglevel = ll == "DEBUG" ? Logging.Debug :
+        ll == "WARN"  ? Logging.Warn  :
+        ll == "ERROR" ? Logging.Error : Logging.Info
+    logger_stream = MPI.Comm_rank(mpicomm) == 0 ? stderr : devnull
+    global_logger(ConsoleLogger(logger_stream, loglevel))
+    @static if haspkg("CUDAnative")
+        device!(MPI.Comm_rank(mpicomm) % length(devices()))
+    end
+
+    polynomialorder = 4
+    base_num_elem = 4
+
+    DFloat = Float64
+    dim = 3
+    
+    brickrange = (range(DFloat(xmin), length=Ne[1]+1, DFloat(xmax)),
+                  range(DFloat(ymin), length=Ne[2]+1, DFloat(ymax)),
+                  range(DFloat(zmin), length=Ne[3]+1, DFloat(zmax)))
+    
+    topl = BrickTopology(mpicomm, brickrange,
+                         periodicity = (false, false, false))
+    dt = 0.00001
+    warpfun = warp_external_topography
+    
+    timeend = dt
+    nsteps = ceil(Int64, timeend / dt)
+    
+    run(mpicomm, ArrayType, dim, topl, warpfun,
+        polynomialorder, timeend, DFloat, dt)
+    
```

Comment:
Can we actually check something (as opposed to having the test just be that the code runs?)

Reply:
other than visualizing the grid, I don't know how to check. unless we save a grid file and compare all the next runs to it.

Reply:
You could compute the norm of the solution or something like this. We have done this [elsewhere](https://github.com/climate-machine/CLIMA/blob/70414b906c2e962a0ca40eba94d14afc47d50fb0/test/DGmethods/Euler/RTB_IMEX.jl#L457-L464) as a test. At least that way if something changes we have some sort of warning that things broke / have changed.

Since you seem to be using an mms solution, you should also be able to [compute error and convergence rates](https://github.com/climate-machine/CLIMA/blob/70414b906c2e962a0ca40eba94d14afc47d50fb0/test/DGmethods/compressible_Navier_Stokes/mms_bc.jl).

Reply:
I will add this check but it may take me a while because I am leaving in two days and I am wrapping other things up. I will get to this though.